Tasks/ServiceFabricUpdateAppVersions/Get-VstsBuild.psm1

```diff
@@ -0,0 +1,87 @@
+﻿function Get-VstsBuild
+{
+    [CmdletBinding()]
+    [OutputType([string])]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $PkgArtifactName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("Specific", "LastSuccessful")]
+        [string]
+        $CompareType,
+
+        [string]
+        $BuildNumber
+    )
+
+    $vstsEndpoint = Get-VstsEndpoint -Name SystemVssConnection -Require
```

Comment:
You might find it easier to use the SDK. e.g. https://github.com/Microsoft/vsts-task-lib/blob/master/powershell/Docs/FullHelp/Get-VstsVssCredentials.md